airbyte-integrations/connectors/source-notion/source_notion/source.py

```diff
@@ -41,7 +41,18 @@ def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) ->
             next(records)
             return True, None
         except requests.exceptions.RequestException as e:
-            return False, e
+            # The most likely user error will be incorrectly configured credentials. We can provide a specific error message for those cases. Otherwise, the stock Notion API message should suffice.
```

Comment:
can you add unit tests which mock the http responses and ensures the check method returns the correct error messages?